client.go

```diff
@@ -26,11 +26,12 @@ func CallWithCodec(cc rpc.ClientCodec, method string, args interface{}, resp int
 	if err := cc.ReadResponseHeader(&response); err != nil {
 		return err
 	}
-	if err := cc.ReadResponseBody(resp); err != nil {
-		return err
-	}
 	if response.Error != "" {
+		cc.ReadResponseBody(nil)
```

Comment:
Added multierror so we don't mask anything. I think that's the best we can do, and multierror should only be needed in the rare cases where we fail emptying the body.